### PR TITLE
Fix ConnectionInfo type (origin -> headers), added typings

### PR DESCRIPTION
**Pull request by tmcgannon**
Fix ConnectionInfo type (in types.ts, 'origin' property was not updated to 'headers').  Added typings.

### CODE REVIEW

**Comment by goriunov** [comment]
Thanks for pull req, could you please also add HTTPS typings (HTTP | HTTPS).

**Comment by tmcgannon** [comment]
Added HTTP | HTTPS.  Also, added types to handleUpgrade() - kind of a hack since you are accessing private net.Socket properties.

File: lib/types.ts [hunk 1]
```diff
@@ -20,7 +22,7 @@ export type ServerConfigs = {
   path?: string,
   port?: number,
   host?: string,
-  server?: any,
+  server?: HTTP.Server,
```
Comment (goriunov) [inline lib/types.ts]: Need to add https here too

**Comment by tmcgannon** [comment]
Sorry for that miss.  HTTPS typing has been added to types.ts.

**Comment by goriunov** [comment]
Looks good to me :) Thank you.

**Pull request merged**